src/sql/parts/query/common/resultSerializer.ts

```diff
@@ -69,7 +69,8 @@ export class ResultSerializer {
 		return self.promptForFilepath(saveRequest).then(filePath => {
 			if (filePath) {
 				let selection: Slick.Range = undefined;
-				if (saveRequest.selection) {
+				// some selection was made
+				if (saveRequest.selection.length > 0) {
```

Comment:
was the null check here  for a reason? Maybe this should be `if (saveRequest.selection && saveRequest.selection.length > 0)`?

Reply:
Selection is never null in the new implementation. It's an empty array if there's none.